File: java/TwoSum.java
// Accepted
import java.util.HashMap;
import java.util.Arrays;
// @lc id=1 lang=java
// problem: two_sum
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> counter = new HashMap();
        for (int i = nums.length - 1; i >= 0; i-- ) {
            counter.putIfAbsent(nums[i], i);
        }
        for (int i = 0; i<nums.length; i++) {
            Integer tmp = counter.get(target - nums[i]);
            if (tmp != null && tmp != i) {
                return new int[] {i, tmp};
            }
        }
        return new int[2];
    }
}
public class TwoSum {
    public static void main(String[] argv) {
        System.out.println(Arrays.toString(new Solution().twoSum(new int[] {3, 2, 4}, 7)));
    }
}

File: go/src/two_sum/two_sum.go
// 2017-11-14 18:54
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, v := range nums {
		m[v] = i
	}
	for i := 0; i < len(nums); i++ {
		j, ok := m[target-nums[i]]
		if ok && i != j {
			return []int{i, j}
		}
	}
	return []int{0, 0}
}

func main() {
	x := []int{3, 2, 4}
	target := 6
	result := twoSum(x, target)
	fmt.Println(result)
}
